app/config/FeatureSwitches.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package config
+
+import play.api.Configuration
+
+import javax.inject.{Inject, Singleton}
+
+/**
+ * Represents a single feature toggle.
+ */
+case class FeatureSwitch(name: String, enabled: Boolean)
+
+/**
+ * Loads all GMP-related feature switches from configuration.
+ */
+@Singleton
+class FeatureSwitches @Inject()(config: Configuration) {
```

Comment:
Is this really needed? We are defining feature switch type here but don't use it anywhere as a type. We still rely on the boolean flags that get resolved in ApplicationConfig as follows

```
def isHipEnabled: Boolean = featureSwitches.hipIntegration.enabled
```

So, wondering if it would just be a simple case of inlining below into ApplicationConfig.
```
config.getOptional[Boolean](s"$prefix.$name").getOrElse(false)
```

If we want to maintain a type for the connector used - lets use it consistently across. We use integration type string so this is not required and a simple inline will do.